sdk/core/azure-core-amqp/inc/azure/core/amqp/internal/connection.hpp

```diff
@@ -181,27 +182,28 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
         ConnectionState oldState)
         = 0;
 
-    /** @brief Called when a new endpoint connects to the connection.
+    /** @brief called when an I/O error has occurred on the connection.
      *
      * @param connection The connection object.
-     * @param endpoint The endpoint that connected.
-     * @return true if the endpoint was accepted, false otherwise.
-     *
-     * @remarks Note that this function should only be overriden if the application is listening
-     * on the connection.
      */
-    virtual bool OnNewEndpoint(Connection const& connection, Endpoint& endpoint)
-    {
-      (void)connection;
-      (void)endpoint;
-      return false;
-    }
+    virtual void OnIOError(Connection const& connection) = 0;
+  };
 
-    /** @brief called when an I/O error has occurred on the connection.
+  class ConnectionEndpointEvents {
+  protected:
+    virtual ~ConnectionEndpointEvents(){};
```

Comment:
Super nit (but it is never a problem, just a matter of preference, if you like it, then append it):
```suggestion
    virtual ~ConnectionEndpointEvents() = default;
```